### PR TITLE
Make metric keys configurable

### PR DESCRIPTION
**Pull request by hamersaw**
# TL;DR
This PR allows manually configuring prometheus metric labels using metric keys. Currently FlytePropeller default metric keys include the workflow and task IDs. This means that as the number of workflows / tasks increases the cardinality of metrics will increase as well. For some deployments this is undesirable. By enabling manual configuration we can tweak metric cardinality to satisfy deployment requirements.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
fixes https://github.com/flyteorg/flyte/issues/2565

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by hamersaw** [comment]
@katrogan did you end up setting the metric keys for single binary after the [flyte admin PR](https://github.com/flyteorg/flyteadmin/pull/429)? It is necessary right?

**Review by EngHabu — APPROVED** [review]

**Review by kumare3 — APPROVED** [review]

**Comment by katrogan** [comment]
@hamersaw so I updated admin to set the metric keys outside of the Serve() entrypoint single binary uses so it wasn't necessary to change anything there (https://github.com/flyteorg/flyte/pull/2526)
